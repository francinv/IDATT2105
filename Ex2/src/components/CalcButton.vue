<script lang="ts">
export default {
  props: {
    label: String,
    action: Function,
    style: String,
  },
};
</script>

<template>
  <button @click="action" :class="style" class="calc-btn">{{ label }}</button>
</template>
