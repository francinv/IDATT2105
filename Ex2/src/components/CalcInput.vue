<script setup lang="ts">
import { storeToRefs } from "pinia";
import { useCalculatorStore } from "@/calculator";

const { firstNumber, secondNumber, isFirstNumberState } = storeToRefs(
  useCalculatorStore()
);
const { calculateFromInput } = useCalculatorStore();
</script>

<script lang="ts"></script>

<template>
  <input
    class="calc-input"
    type="text"
    @change="(event) => calculateFromInput(event.target.value)"
    :value="isFirstNumberState ? firstNumber : secondNumber"
  />
</template>
