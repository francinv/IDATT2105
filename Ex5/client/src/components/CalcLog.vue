<script setup lang="ts">
import { storeToRefs } from "pinia"
import { useStore } from "@/store"

const { log } = storeToRefs(useStore())
</script>

<script lang="ts">
import { defineComponent } from "vue"

export default defineComponent({
  data() {
    return {
      showLog: true,
    }
  },
  methods: {
    handleShow() {
      this.showLog = !this.showLog
    },
  },
})
</script>

<template>
  <div class="log">
    <div class="log-header">
      <h2>Logg over utfÃ¸rte kalkulasjoner</h2>
      <button class="log-btn" @click="handleShow">
        {{ showLog ? "Skjul ðŸ‘€" : "Vis ðŸ‘€" }}
      </button>
    </div>
    <ol v-if="showLog">
      <li v-for="element in log" :key="element">{{ element }}</li>
    </ol>
  </div>
</template>
