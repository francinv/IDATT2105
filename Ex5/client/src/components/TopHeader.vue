<script lang="ts">
import authClient from "@/server/auth"
import { ref } from "vue"

export default {
  setup() {
    const isLoggedIn = ref(false)

    const checkLogin = async () => {
      const res = await authClient.isTokenValid()
      isLoggedIn.value = res
    }

    checkLogin()
    return {
      isLoggedIn,
    }
  },
}
</script>

<template>
  <header>
    <img alt="Calculator logo" class="logo" src="@/assets/logo.png" width="100" height="100" />
    <nav>
      <ul>
        <li><router-link to="/" active-class="link-active" v-if="isLoggedIn">Hjem</router-link></li>
        <li><router-link to="/contact" active-class="link-active" v-if="isLoggedIn">Kontakt</router-link></li>
        <li><router-link to="/profile" active-class="link-active" v-if="isLoggedIn">Profil</router-link></li>
        <li><router-link to="/login" active-class="link-active" v-if="!isLoggedIn">Logg inn</router-link></li>
        <li><router-link to="/register" active-class="link-active" v-if="!isLoggedIn">Registrer deg</router-link></li>
      </ul>
    </nav>
  </header>
</template>
