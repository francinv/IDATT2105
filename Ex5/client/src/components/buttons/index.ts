export { default as BaseButton } from "./Button.vue"
export { default as CalculatorButton } from "./CalcButton.vue"
