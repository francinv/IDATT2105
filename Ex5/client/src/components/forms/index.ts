export { default as CalcInput } from "./CalcInput.vue"
export { default as TextArea } from "./TextArea.vue"
export { default as TextInput } from "./TextInput.vue"
