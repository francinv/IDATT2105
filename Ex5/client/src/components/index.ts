export * from "./buttons";
export * from "./forms";

export { default as CalcLog } from "./CalcLog.vue";
export { default as Calculator } from "./Calculator.vue";
