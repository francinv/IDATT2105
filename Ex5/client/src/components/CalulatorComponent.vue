<script setup lang="ts">
import { CalcInput, CalculatorButton } from "@/components"
import { useStore } from "@/store"
import { storeToRefs } from "pinia"

const { showErrorMessage } = storeToRefs(useStore())
const { equals, negate, clear, divide, multiply, del, subtract, addition, decimal, setNumber } = useStore()
</script>

<template>
  <div class="calculator-container">
    <div>
      <CalcInput />
    </div>
    <div class="button-container">
      <CalculatorButton label="AC" :action="clear" class="dark-gray" />
      <CalculatorButton label="+/-" :action="negate" class="dark-gray" />
      <CalculatorButton label="DEL" :action="del" class="dark-gray" />
      <CalculatorButton label="÷" :action="divide" class="orange" />
      <CalculatorButton label="7" :action="() => setNumber(7)" class="light-gray" />
      <CalculatorButton label="8" :action="() => setNumber(8)" class="light-gray" />
      <CalculatorButton label="9" :action="() => setNumber(9)" class="light-gray" />
      <CalculatorButton label="x" :action="multiply" class="orange" />
      <CalculatorButton label="4" :action="() => setNumber(4)" class="light-gray" />
      <CalculatorButton label="5" :action="() => setNumber(5)" class="light-gray" />
      <CalculatorButton label="6" :action="() => setNumber(6)" class="light-gray" />
      <CalculatorButton label="-" :action="subtract" class="orange" />
      <CalculatorButton label="1" :action="() => setNumber(1)" class="light-gray" />
      <CalculatorButton label="2" :action="() => setNumber(2)" class="light-gray" />
      <CalculatorButton label="3" :action="() => setNumber(3)" class="light-gray" />
      <CalculatorButton label="+" :action="addition" class="orange" />
      <CalculatorButton label="0" :action="() => setNumber(0)" class="light-gray left radius-left" />
      <CalculatorButton label="," :action="decimal" class="light-gray right" />
      <CalculatorButton label="=" :action="equals" class="orange right-last radius-right" />
    </div>
    <div v-if="showErrorMessage" class="error">
      <h1>‼️</h1>
      <h3>Ser ut som det er bokstaver i feltet. Vennligst fjern og prøv på nytt.</h3>
    </div>
  </div>
</template>
