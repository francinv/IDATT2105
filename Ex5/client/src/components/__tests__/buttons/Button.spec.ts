import { afterEach, beforeEach, describe, expect, it } from "vitest"
import { mount, VueWrapper } from "@vue/test-utils"
import { BaseButton } from "@/components"

describe("BaseButton unit test", () => {
  let wrapper: VueWrapper | null = null

  beforeEach(() => {
    wrapper = mount(BaseButton)
    wrapper.setProps({
      label: "Test",
      disabled: false,
      action: () => console.log("Test"),
    })
  })

  afterEach(() => {
    wrapper?.unmount()
  })

  it("should render the component", () => {
    if (!wrapper) return
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toBe("Test")
    expect(wrapper.props("label")).toBe("Test")
    expect(wrapper.props("disabled")).toBe(false)
    expect(wrapper.props("action")).toBeInstanceOf(Function)
  })

  it("should emit a click event", async () => {
    if (!wrapper) return
    await wrapper.trigger("click")
    expect(wrapper.emitted("click")).toBeTruthy()
  })
})
