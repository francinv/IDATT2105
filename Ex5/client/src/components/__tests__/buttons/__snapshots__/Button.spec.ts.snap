// Vitest Snapshot v1

exports[`BaseButton unit test > should render the component 1`] = `"<div class=\\"btn-container\\"><button type=\\"button\\" class=\\"\\">Test</button></div>"`;
